src/Stormpath/Client.php

```diff
@@ -100,15 +105,21 @@ class Client extends Magic
      * @param $baseUrl optional parameter for specifying the base URL when not using the default one
      *         (https://api.stormpath.com/v1).
      */
-    public function __construct(ApiKey $apiKey, $cacheManager, $cacheManagerOptions, $baseUrl = null)
+    public function __construct(ApiKey $apiKey, $cacheManager, $cacheManagerOptions, $baseUrl = null, $authenticationScheme = null)
```

Comment:
Instead of the authentication scheme, we should have a signer here so we don't have to resolve it from the client. It should be resolved from the `ClientBuilder`.